### PR TITLE
feat: triple support self exception

### PR DESCRIPTION
**Pull request by aamingaa**
The triple support self defines exception

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package org.apache.dubbo.rpc.protocol.tri;
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java:1]: Please add ASF license header for all newly created files

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+//        byte[] tripleExceptionWrapperBytes = tripleExceptionWrapperUtils.packRequest(exception);
+//        CharSequence msg = StreamUtils.encodeBase64ASCII(tripleExceptionWrapperBytes);
+//        msg = TriRpcStatus.encodeMessage((String) msg);
+//        msg = TriRpcStatus.decodeMessage(msg.toString());
+//        byte[] tripleExceptionWrapperDecodeBytes = StreamUtils.decodeASCIIByte(msg);
+////        TripleExceptionWrapperUtils tripleExceptionWrapperUtils2 = TripleExceptionWrapperUtils.init(new ScoreException(), url);
+//        Object object1 = tripleExceptionWrapperUtils.unPackRequest(tripleExceptionWrapperDecodeBytes);
+//        System.out.println(object1 instanceof ScoreException);
+//        System.out.println(object1);
+//    }
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java:130]: remove unused code

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleClientStream.java [hunk 3]
```diff
@@ -283,7 +286,23 @@ private TriRpcStatus statusFromTrailers(Http2Headers trailers) {
...
+                if (messageException != null) {
+                    String decodeMessageException = TriRpcStatus.decodeMessage(messageException.toString());
+                    byte[] decodeMessageExceptionByte = StreamUtils.decodeASCIIByte(decodeMessageException);
+                    TripleExceptionWrapperUtils tripleExceptionWrapperUtils = TripleExceptionWrapperUtils.
+                        init(null, URL.valueOf("TEST"));
+                    Object messageExceptionObj = null;
+                    try {
+                        messageExceptionObj = tripleExceptionWrapperUtils.unPackRequest(decodeMessageExceptionByte);
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleClientStream.java:300]: use logger

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleClientStream.java [hunk 4]
```diff
@@ -283,7 +286,23 @@ private TriRpcStatus statusFromTrailers(Http2Headers trailers) {
...
             TriRpcStatus status = intStatus == null ? null : TriRpcStatus.fromCode(intStatus);
             if (status != null) {
-                final CharSequence message = trailers.get(TripleHeaderEnum.MESSAGE_KEY.getHeader());
+                CharSequence message = trailers.get(TripleHeaderEnum.MESSAGE_KEY.getHeader());
+                CharSequence messageException = trailers.get(TripleHeaderEnum.EXCEPTION_KEY.getHeader());
+                if (messageException != null) {
+                    String decodeMessageException = TriRpcStatus.decodeMessage(messageException.toString());
+                    byte[] decodeMessageExceptionByte = StreamUtils.decodeASCIIByte(decodeMessageException);
+                    TripleExceptionWrapperUtils tripleExceptionWrapperUtils = TripleExceptionWrapperUtils.
+                        init(null, URL.valueOf("TEST"));
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleClientStream.java:295]: Should this utils be a field of the stream?

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleServerStream.java [hunk 5]
```diff
@@ -21,13 +21,11 @@
 import org.apache.dubbo.common.logger.Logger;
 import org.apache.dubbo.common.logger.LoggerFactory;
 import org.apache.dubbo.common.utils.StringUtils;
-import org.apache.dubbo.rpc.HeaderFilter;
-import org.apache.dubbo.rpc.Invoker;
-import org.apache.dubbo.rpc.PathResolver;
-import org.apache.dubbo.rpc.TriRpcStatus;
+import org.apache.dubbo.rpc.*;
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleServerStream.java:24]: do not use comma import

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleServerStream.java [hunk 6]
```diff
@@ -190,6 +189,19 @@ private Http2Headers getTrailers(TriRpcStatus rpcStatus, Map<String, Object> att
...
+        TripleExceptionWrapperUtils tripleExceptionWrapperUtils = TripleExceptionWrapperUtils.init(throwable, url);
+        try {
+            byte[] exceptionBytes = tripleExceptionWrapperUtils.packRequest(throwable);
+            String exceptionMessage = StreamUtils.encodeBase64ASCII(exceptionBytes);
+            exceptionMessage = TriRpcStatus.encodeMessage(exceptionMessage);
+            headers.set(TripleHeaderEnum.EXCEPTION_KEY.getHeader(),
+                exceptionMessage);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleServerStream.java:203]: same

File: dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/support/IGreeterException.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package org.apache.dubbo.rpc.protocol.tri.support;
+
+public class IGreeterException extends Exception {
+        //异常信息
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/support/IGreeterException.java:4]: comment in eng

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java (hunk 1 above)
Comment (sunchaser-lilu) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java:1]: > Please add ASF license header for all newly created files

Maybe we can use apache/skywalking-eyes/header@main to automatically add ASF license header to newly created files in GitHub action. I'm willing to submit a PR if needed, thanks.

Comment (AlbumenJ) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleExceptionWrapperUtils.java:1]: > > Please add ASF license header for all newly created files
> 
> Maybe we can use apache/skywalking-eyes/header@main to automatically add ASF license header to newly created files in GitHub action. I'm willing to submit a PR if needed, thanks.

@sunchaser-lilu go ahead~

**Pull request closed without merging**